Patch by Robert Scheck <redhat@linuxnetz.de> for slang >= 1.4.9, which 
makes slang rebuildable using gcc 4.

--- slang-1.4.9/src/sltoken.c		2003-03-23 08:06:40.000000000 +0100
+++ slang-1.4.9/src/sltoken.c.gcc4	2005-03-14 09:01:16.000000000 +0100
@@ -789,7 +789,7 @@
 #if _SLANG_HAS_DEBUG_CODE
 	     tok->line_number++;
 #endif
-	     Input_Line = LLT->read(LLT);
+	     Input_Line = (LLT->read) (LLT);
 	     if ((NULL == Input_Line) || SLang_Error)
 	       {
 		  Input_Line_Pointer = Input_Line = NULL;
@@ -1185,7 +1185,7 @@
 	  {
 	   case '\n':
 	   case 0:
-	     if (NULL == (Input_Line = LLT->read(LLT)))
+	     if (NULL == (Input_Line = (LLT->read) (LLT)))
 	       {
 		  Input_Line_Pointer = Input_Line = NULL;
 		  return;
--- slang-1.4.9/src/slposio.c		2003-03-23 08:06:40.000000000 +0100
+++ slang-1.4.9/src/slposio.c.gcc4	2005-03-14 09:03:00.000000000 +0100
@@ -166,7 +166,7 @@
        || (NULL == (b = SLmalloc (len + 1))))
      goto return_error;
    
-   if (-1 == f->read (f->fd, b, &len))
+   if (-1 == (f->read) (f->fd, b, &len))
      {
 	_SLerrno_errno = errno;
 	goto return_error;
